File: tests/verify_server_imports.py
import sys
import os
from pathlib import Path

BASE_DIR = Path(__file__).parent.resolve()
sys.path.insert(0, str(BASE_DIR))


def test_import(module_name):
    """Test import."""
    try:
        # [X-Ray auto-fix] print(f"Testing import: {module_name}...", end="", flush=True)
        __import__(module_name)
        print(" OK")
        return True
    except Exception:
        # [X-Ray auto-fix] print(f" FAILED: {e}")
        import traceback

        traceback.print_exc()
        return False


print("\n--- ZenAI Deep Import Verification ---\n")

modules = [
    "config_system",
    "utils",
    "nicegui",
    "zena_mode.server",
    "model_manager",
    "voice_service",
    "async_backend",
    "security",
]

all_ok = True
for m in modules:
    if not test_import(m):
        all_ok = False

if all_ok:
    print("\n✅ All core modules import successfully.")
else:
    print("\n❌ some imports FAILED. Check the errors above.")

File: rust_output/src/verify_server_imports.rs
use anyhow::{Result, Context};

pub static BASE_DIR: std::sync::LazyLock<String /* Path.parent().unwrap_or(std::path::Path::new("")).resolve */> = std::sync::LazyLock::new(|| Default::default());

pub static MODULES: std::sync::LazyLock<Vec<String>> = std::sync::LazyLock::new(|| Vec::new());

pub const ALL_OK: bool = true;

/// Test import.
pub fn test_import(module_name: String) -> Result<()> {
    // Test import.
    // try:
    {
        println!("Testing import: {}...", module_name);
        __import__(module_name);
        println!("{}", " OK".to_string());
        true
    }
    // except Exception as _e:
}
